mlflow/pipelines/pipeline.py

```diff
@@ -180,7 +180,7 @@ def _resolve_pipeline_steps(self) -> List[BaseStep]:
 
     @experimental
     @abc.abstractmethod
-    def get_artifact(self, artifact_name: str):
+    def get_artifact(self, artifact_name: str, get_path: bool = False):
```

Comment:
I don't like changing this API to return either an object or a path depending on an input parameter.

Can we add a separate function like `get_artifact_path` to specifically address this needs?

Reply:
yea i explored this option originally. i believe that there will be a lot of duplicated code with a separate function `get_artifact_path`.

cc: @dbczumar

Reply:
No strong preference here.